framework/src/timesteppers/CSVTimeSequenceStepper.C

```diff
@@ -0,0 +1,69 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "CSVTimeSequenceStepper.h"
+#include "DelimitedFileReader.h"
+
+template <>
+InputParameters
+validParams<CSVTimeSequenceStepper>()
+{
+  InputParameters params = validParams<TimeSequenceStepperBase>();
+  params.addRequiredParam<FileName>("file_name",
+                                    "name of the file in which the time sequence is read");
+  params.addParam<bool>(
+      "header", true, "indicates whether the file contains a header with the column names");
+  params.addParam<std::string>("delimiter", ",", "delimiter used to parse the file");
+  params.addParam<std::string>(
+      "column_name", "time", "name of the column which contains the time sequence");
+  params.addRangeCheckedParam<int>(
+      "column_index", "column_index >= 0", "index of the column which contains the time sequence");
+  params.addClassDescription(
+      "Solves the Transient problem at a sequence of given time points read in a file.");
+  return params;
+}
+
+CSVTimeSequenceStepper::CSVTimeSequenceStepper(const InputParameters & parameters)
+  : TimeSequenceStepperBase(parameters),
+    _file_name(getParam<FileName>("file_name")),
+    _header(getParam<bool>("header")),
+    _delimiter(getParam<std::string>("delimiter")),
+    _column_name(getParam<std::string>("column_name")),
+    _column_index(isParamValid("column_index") ? getParam<int>("column_index") : -1)
+{
+}
+
+void
+CSVTimeSequenceStepper::init()
+{
+  MooseUtils::DelimitedFileReader file(_file_name, _header, _delimiter);
+  file.read();
+
+  std::vector<Real> instants;
+
+  if (_column_index >= 0)
```

Comment:
Could compare against the numeric_limits.

Reply:
I've added a _search_by_index boolean (set to true in the user has requested an index rather than a name). I've also consted the _header boolean.